include/simics/dmllib.h

```diff
@@ -1221,6 +1219,454 @@ _deserialize_simple_event_data(
     return error;
 }
 
+// All auxiliary info needed to de/serialize suspended calls created
+// from a specific after-on-hook.
+//
+// An after-on-hook is unique in respect to three things:
```

Comment:
Do we define the term `after-on-hook` somewhere? We use it informally in speech, but perhaps we should find a clearer term to distinguish caller from callee, like "after hook statement"?

Reply:
The eternal debate! Do we call them "after-on-hook statements" or "hook-based after statements"! And I'm still not sure which I prefer. Even if we have a long, formal name I want to have the option for a shorter informal name: "after-on-hook" allows me to e.g. say "illegal after statement on hook {hookexpr}" in error messages, which would instead become "illegal after statement based on hook {hookexpr}", which I honestly find rather terrible.

Reply:
I think your "after statement on hook" example is a clear expression that stands on its own, it's clear what it means. What I'm afraid of about `after-on-hook` is that it feels like a label of the feature added by this PR, rather than a way to disambiguate it  from other `after` statements. We didn't introduce terms `after-on-time` and `after-on-cycles` for  `after 7 s` and `after 7 cycles`.

Reply:
True, but that's because we morally considered them of the same form (despite the fact they use different API call), and we only had that form until now: _after-delay_. Now that we're introducing new forms with _completely_ different semantics, we _need_ to qualify the different forms of after to separate them. Indeed, in the immediate after PR I refer to `after ... s` and `after ... cycles` as "After Delay Statements".